card/mod/05_standard/spec/set/type/search_type_spec.rb

```diff
@@ -35,6 +35,41 @@
     expect(pointer_card.content).to eq(%{{"type":"User"}})
   end
 
+  context 'with right plus array' do
+    it 'render the card list and paging correctly' do
+      Card.create! name: 'Samsung'
+      Card.create! name: 'Samsung+tag'
+      Card.create! name: 'Samsung+source'
+      Card.create! name: 'Apple'
+      Card.create! name: 'Apple+tag'
+      Card.create! name: 'Apple+source'
+      Card.create! name: 'HTC'
+      Card.create! name: 'HTC+tag'
+      Card.create! name: 'HTC+source'
+      search_card = Card.create!(
+        name: 'search_with_right_plus',
+        type_id: Card::SearchTypeID,
+        content: %{
+            {
+              "right_plus":{
+                "name":["in","tag","source"]
+              },
+              "limit":1
+            }
+        }
+      )
+      html = search_card.format.render_open
+      expect(html).to have_tag('ul', with: { class: 'pagination paging' }) do
+        with_tag 'a', text: '1'
+        with_tag 'a', text: '2'
+        with_tag 'a', text: '3'
+      end
+      expect(html).to have_tag('div', with: { class: 'search-result-list' }) do
+        with_tag 'div', with: { id: 'Apple' }
+      end
```

Comment:
The problem won't happen in a normal query. 
The function call`Card.search(s)` in 
```
def item_cards params={}
  s = query(params)
  raise('OH NO.. no limit') unless s[:limit]
  # forces explicit limiting
  # can be 0 or less to force no limit
  Card.search(s)
end
```
will affect the `@query`'s params.

For example in wikirate's source filter,
part of the `@query` will be changed from 
`:right_plus=>[{"name"=>["in", "tag", "company", "topic"]` to 
`:name=>["tag", "company", "topic"]`.

This affect the further query of count for paging.